### PR TITLE
Refactor controller functions to use service layer for retrieving user data (/users/userId/:userId) route (/users/self) route

### PR DESCRIPTION
**Pull request by mailmesriza98**
This closes https://github.com/Real-Dev-Squad/website-backend/issues/1235 and https://github.com/Real-Dev-Squad/website-backend/issues/1236

(/users/userId/:userId) route
(/users/self) route

251230948-c188936a-b556-43cd-967d-2e0846da0984

### CODE REVIEW

File: controllers/users.js [hunk 1]
```diff
@@ -51,15 +50,7 @@ const getUserById = async (req, res) => {
...
 
-  const { phone = "", email = "", ...user } = result.user;
-  try {
-    user.phone = obfuscate.obfuscatePhone(phone);
-    user.email = obfuscate.obfuscateMail(email);
-  } catch (error) {
-    logger.error(`Error while formatting phone and email: ${error}`);
-    return res.boom.badImplementation("Error while formatting phone and email");
-  }
-
```
Comment (prakashchoudhary07) [inline controllers/users.js:62]: Please confirm if, the phone(88xxxxxxxx) and email (sommailxxxxxxxx.xx) being removed will break any frontend app, where it's being used

File: controllers/users.js [hunk 2]
```diff
@@ -133,6 +124,7 @@ const getUsers = async (req, res) => {
 
 const getUser = async (req, res) => {
   try {
+    // replace below line woth call to retrieveUsers(id)
```
Comment (prakashchoudhary07) [inline controllers/users.js:127]: If its a todo comment then please add a TODO prefix to it and create a new issue for it

**Review by fakhruddinkw — COMMENTED** [review]
Are we retrieving single user or multiple here?

**Review by bksh05 — APPROVED** [review]
LGTM

**Comment by bharati-21** [comment]
Please attach coverage

File: controllers/users.js [hunk 3]
```diff
@@ -51,15 +50,7 @@ const getUserById = async (req, res) => {
...
-  const { phone = "", email = "", ...user } = result.user;
-  try {
-    user.phone = obfuscate.obfuscatePhone(phone);
-    user.email = obfuscate.obfuscateMail(email);
-  } catch (error) {
-    logger.error(`Error while formatting phone and email: ${error}`);
-    return res.boom.badImplementation("Error while formatting phone and email");
-  }
-
+  const user = result.user;
```
Comment (iamitprakash) [inline controllers/users.js:53]: are you sure, result can't be null here in any edge case..?

**Review by iamitprakash — CHANGES_REQUESTED** [review]

**Review by iamitprakash — APPROVED** [review]

**Review by heyrandhir — APPROVED** [review]

**Pull request merged**